### DISCUSSION TITLE
post_trial_gap blocked the programme

### DISCUSSION THREAD

**Discussion by ZifengLIU143** [jspsych/jsPsych#2778] [Q&A] [answered]
Hi, 
when I add a `post_trial_gap` between two trials, the programme is blocked. It does not work anymore. But it works well when I specify the `trial_duration`. This is my code, could you please have a look at it? Perhaps I misunderstood some fundamental principles behind these two parameters. Thanks!

    `var test = {
      type:jsPsychAudioKeyboardResponse,
      stimulus: function(){
        return jsPsych.timelineVariable("pathFile")},
      choices: "NO_KEYS",
      post_trial_gap: 500
    };

    var test1 = {
      type: jsPsychHtmlButtonResponse,
      stimulus: "",
      choices:['0','1','2','3','4','5','6','7','8','9','Continue']
    };

    /* define test procedure */
    var test_procedure = {
      timeline: [test, test1],
      timeline_variables: demostimuli,
      randomize_order: true,
    };

    timeline.push(test_procedure)`

**Comment by jodeleeuw** [comment, marked as answer]
Hi @ZifengLIU143,

In your audio trial, there is no duration set on the length of the trial, so the trial lasts indefinitely because no keyboard responses are allowed. If you want the trial to end after the audio finishes playing, you can use the `trial_ends_after_audio: true` parameter.

```js
var test = {
  type:jsPsychAudioKeyboardResponse,
  stimulus: function(){
    return jsPsych.timelineVariable("pathFile")},
  choices: "NO_KEYS",
  trial_ends_after_audio: true,
  post_trial_gap: 500
};
```

**Reply by ZifengLIU143** [reply]
Thank you very much! It works now!
